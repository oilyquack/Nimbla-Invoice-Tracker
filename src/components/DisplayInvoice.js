import React from "react";

class DisplayInvoice extends React.Component {
  constructor(props) {
    super(props);

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(event) {
    event.preventDefault();

    this.props.editReceiver("edit");
  }

  render() {
    return (
      <div>
        <h2>Invoice #: {this.props.invoiceId}</h2>
        <p>Â£{this.props.amount}</p>
        <p>Pay By: {this.props.payBy}</p>
        <p>Customer: {this.props.customer}</p>
        <p>Services: {this.props.services}</p>
        <button onClick={this.handleClick}>Edit</button>
      </div>
    );
  }
}

export default DisplayInvoice;
