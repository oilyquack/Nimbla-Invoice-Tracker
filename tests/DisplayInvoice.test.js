import React from "react";
import renderer from "react-test-renderer";
import DisplayInvoice from "../src/components/DisplayInvoice";

it("matches the snapshot", () => {
  const tree = renderer.create(<DisplayInvoice />);
  expect(tree).toMatchSnapshot();
});
