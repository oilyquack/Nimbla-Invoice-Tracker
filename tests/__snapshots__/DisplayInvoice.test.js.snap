// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="app__invoices__display--read"
>
  <label>
    Invoice #
  </label>
  <p>
    <span
      className="display-title"
    >
      Amount:
    </span>
     Â£
  </p>
  <p>
    <span
      className="display-title"
    >
      Pay By:
    </span>
     
  </p>
  <p>
    <span
      className="display-title"
    >
      Customer:
    </span>
     
  </p>
  <p>
    <span
      className="display-title"
    >
      Services:
    </span>
     
  </p>
  <span
    className="display-title"
  >
    UNPAID
  </span>
  <button
    onClick={[Function]}
  >
    Edit
  </button>
</div>
`;
